src/components/TextField/TextArea.js

```diff
@@ -0,0 +1,276 @@
+import StyleObjectPropType from '../../prop-types/style';
+import Tooltip from '../Tooltip/Tooltip';
+import TooltipBox from '../Tooltip/TooltipBox';
+import { TooltipBoxStatus, TooltipPosition } from '../Tooltip/TooltipEnums';
+import IconRequired from '../Icon/IconRequired';
+import './TextArea.scss';
+
+const propTypes = {
+    className: PropTypes.string,
+    componentRef: PropTypes.func,
+    hasAutoHeight: PropTypes.bool,
+    hasLabelAlways: PropTypes.bool,
+    isDisabled: PropTypes.bool,
+    isFullWidth: PropTypes.bool,
+    isReadOnly: PropTypes.bool,
+    isRequired: PropTypes.bool,
+    isValid: PropTypes.bool,
+    label: PropTypes.string,
+    name: PropTypes.string,
+    onBlur: PropTypes.func,
+    onChange: PropTypes.func,
+    onEnterKey: PropTypes.func,
+    onFocus: PropTypes.func,
+    onKeyDown: PropTypes.func,
+    onKeyPress: PropTypes.func,
+    onKeyUp: PropTypes.func,
+    placeholder: PropTypes.string,
+    rows: PropTypes.number,
+    style: StyleObjectPropType(),
+    tooltipError: PropTypes.oneOfType([
+        PropTypes.element,
+        PropTypes.string,
+    ]),
+    tooltipHint: PropTypes.oneOfType([
+        PropTypes.element,
+        PropTypes.string,
+    ]),
+    tooltipRequired: PropTypes.oneOfType([
+        PropTypes.element,
+        PropTypes.string,
+    ]),
+    value: PropTypes.string,
+};
+
+const defaultProps = {
+    className: null,
+    componentRef: null,
+    hasAutoHeight: false,
+    hasLabelAlways: false,
+    isDisabled: false,
+    isFullWidth: false,
+    isReadOnly: false,
+    isRequired: false,
+    isValid: null,
+    label: null,
+    name: null,
+    onBlur: null,
+    onChange: null,
+    onEnterKey: null,
+    onFocus: null,
+    onKeyDown: null,
+    onKeyPress: null,
+    onKeyUp: null,
+    placeholder: null,
+    rows: 1,
+    style: null,
+    tooltipError: null,
+    tooltipHint: null,
+    tooltipRequired: 'required',
+    value: '',
+};
+
+let lastInstanceId = 0;
+
+class TextArea extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            hasFocus: false,
+            hasMouseOver: false,
+            showTooltip: false,
+            value: props.value,
+        };
+
+        lastInstanceId += 1;
+        this.uid = `textarea-${lastInstanceId}`;
+    }
+
+    componentDidMount = () => {
+        autosize(this.inputRef);
+    }
+
+    handleMouseEnter = () => {
+        this.setState({ hasMouseOver: true });
+    }
+
+    handleMouseLeave = () => {
+        this.setState({ hasMouseOver: false });
+    }
+
+    handleInputRef = (ref) => {
+        const { componentRef } = this.props;
+        this.inputRef = ref;
+        if (componentRef) {
+            componentRef(ref);
+        }
+    }
+
+    handleInputBlur = (event) => {
+        const onBlur = this.props.onBlur || (() => {});
+        this.setState({
+            hasFocus: false,
+            showTooltip: false,
+        }, () => {
+            onBlur(event);
```

Comment:
We need to handle the `onBlur` (or other callbacks) throwing an error. There is no defined behaviour for this.

Reply:
Can we discuss this please?